applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -0,0 +1,96 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "solving_strategies/strategies/solving_strategy.h"
+#include "spaces/ublas_space.h"
+#include "linear_solvers/linear_solver.h"
+#include "solving_strategies/convergencecriterias/mixed_generic_criteria.h"
+#include "solving_strategy_factory.hpp"
+#include "factories/standard_linear_solver_factory.h"
+#include "solving_strategies/strategies/residualbased_linear_strategy.h"
+#include "custom_strategies/strategies/geo_mechanics_newton_raphson_strategy.hpp"
+#include "scheme_factory.hpp"
+#include "convergence_criteria_factory.hpp"
+#include "builder_and_solver_factory.hpp"
+
+namespace Kratos
+{
+
+template<class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class SolvingStrategyFactory
+{
+public:
+    [[nodiscard]] static std::unique_ptr<SolvingStrategy<TSparseSpace, TDenseSpace>> Create(Parameters& rSolverSettings, ModelPart& rModelPart)
+    {
+        const std::string strategy_type = "strategy_type";
+
+        KRATOS_ERROR_IF_NOT(rSolverSettings.Has(strategy_type))
+        << "The parameter strategy_type is undefined, aborting.";
+
+        auto solver = LinearSolverFactory<TSparseSpace, TDenseSpace>().Create(rSolverSettings["linear_solver_settings"]);
+        auto scheme = SchemeFactory<TSparseSpace, TDenseSpace>::Create(rSolverSettings);
+        auto builder_and_solver = BuilderAndSolverFactory<TSparseSpace, TDenseSpace, TLinearSolver>::Create(rSolverSettings, solver);
+        auto criteria = ConvergenceCriteriaFactory<TSparseSpace, TDenseSpace>::Create(rSolverSettings);
+
+        Parameters parameters = ExtractStrategyParameters(rSolverSettings);
+
+        if (rSolverSettings[strategy_type].GetString() ==
+            GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::Name())
+        {
+            return std::make_unique<GeoMechanicsNewtonRaphsonStrategy<TSparseSpace,
+                    TDenseSpace,
+                    TLinearSolver>>(rModelPart,
+                    scheme,
+                    solver,
+                    criteria,
+                    builder_and_solver,
+                    parameters);
```

Comment:
I would suggest to limit the scope of `parameters` to this if-clause. In fact, a temporary will do:
```suggestion
                    ExtractStrategyParameters(rSolverSettings));
```
(And remove the definition `parameters` above.)

Reply:
Yes, makes it a bit more readable, agreed